### PR TITLE
docs: Initial draft of Builder style guide.

### PR DESCRIPTION
**Pull request by peter-b**
[Rendered version](https://github.com/peter-b/livecode-ide/blob/docs-lcbstyleguide/Documentation/guides/lcb-style-guide.md)

### CODE REVIEW

**Comment by peter-b** [comment]
Pretty basic at the moment -- probably still lots of additional content needed.

File: Documentation/guides/lcb-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+## Naming
+
+### Module name
+
+The module name uses reverse DNS notation.  For example, a module created by
+the Example Organisation would use module names beginning with `org.example`.
+
+Replace any hyphen (`-`) characters in a domain name with underscore (`_`)
+characters.  For example, a module derived from the `fizz-buzz.example.org`
+domain could be `org.example.fizz-buzz`.
```
Comment (livecodefraser) [inline Documentation/guides/lcb-style-guide.md]: s/-/_/

**Comment by peter-b** [comment]
The "How to write in-line documentation" guide should probably be moved into here.

File: Documentation/guides/lcb-style-guide.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+Copyright 2015 LiveCode Ltd.
+
+[TOC]
+
+## Introduction
+
+This document describes best practices to be followed when working with
+LiveCode Builder source code.  Please follow it *especially* when writing code
+to be included with the main LiveCode product distributable.
+
```
Comment (livecodestephen) [inline Documentation/guides/lcb-style-guide.md:11]: Is "distributable" really a word!  I don't think it's needed here anyway.

Comment (peter-b) [inline Documentation/guides/lcb-style-guide.md:11]: Yes, it is really a word. I'll replace it with "package", however.

Comment (livecodestephen) [inline Documentation/guides/lcb-style-guide.md:11]: I suppose I really meant is it a noun. :smile:

File: Documentation/guides/lcb-style-guide.md (hunk 1 above)
Comment (runrevmark) [inline Documentation/guides/lcb-style-guide.md]: If the domain starts with digit then prefix with '_'

**Pull request merged**